code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -133,9 +133,16 @@
 	return ..()
 
 /**
- * Handler for `COMSIG_OBJ_HIDE`, connects only if `hide` is set to `TRUE`. Calls `update_cap_visuals` on pipe and its connected nodes
+ * Ran on `Initialize` if hide is `TRUE`, allows subtypes to register their own hiding logic.
+ * Including registering cap updating checks to run on different signals.
  */
-/obj/machinery/atmospherics/proc/on_hide(datum/source, underfloor_accessibility)
+/obj/machinery/atmospherics/proc/setup_hiding()
+	RegisterSignal(src, COMSIG_OBJ_HIDE, PROC_REF(on_hide))
+
+/**
+ * Handler for signals registered in `setup_hiding`, connects only if `hide` is set to `TRUE`. Calls `update_cap_visuals` on pipe and its connected nodes
```

Comment:
same sort of applies here, what is going on is semi clear, the why is less so (though it isn't your work so I'm not gonna make you change it)